dist/actions/entrypoint.sh

```diff
@@ -77,8 +77,14 @@ if [ ! -z "$GOOGLE_CREDENTIALS" ]; then
     gcloud auth activate-service-account --key-file=$GCLOUD_KEYFILE
 fi
 
-# Next, lazily install packages if required.
-if [ -e package.json ] && [ ! -d node_modules ]; then
+# Next, run npm install. We always call this, as
```

Comment:
It's likely that this is the change we'd like to take, but can you help me understand why this is needed? Something doesn't quite sound right here. But it could just be that I'm misunderstanding the comment.

If Pulumi will always run the "npm scripts that install plugins", then why is this needed exactly? Is it that Pulumi running the "install plugins" step is skipped if packages are found, but in the case of running during GitHub Actions the plugins are located elsewhere and therefore unavailable?

If we expect `npm install` to be a noop if the `package-lock.json` file is present, then would it also mean that `npm install` will skip installing any required plugins?

A quick read of the [documentation for npm-install](https://docs.npmjs.com/cli/install) unfortunately doesn't help any.

Can on confirm some of the finer details here, and then update the comment to make it clearer and more concise? It's possible that this is the exact functional change we need to make, but as described in the comment it doesn't quite seem right.

Reply:
@chrsmith Pulumi installs plugins to ~/.pulumi. This installation doesn't respect $HOME environment variable, it uses `getent`, which gets the home from `/etc/passwd`.

So when I run `npm install` (this could also be a pulumi login or pulumi preview command) manually, some post-install script fetches the required plugins; Pulumi won't run `npm install`, hence fetching the plugins, when `node_modules` exists.

So by the time I reach `pulumi up`, `node_modules` exists and the plugins do not; as that directory isn't persisted by GitHub Actions across jobs.